MIDI Editor/talagan_OneSmallStep.lua

```diff
@@ -0,0 +1,395 @@
+
+  # Reaper forum thread
+
+    The forum thread does not exist yet (at release time). Please search "One Small Step" on reaper forums for now (until a new version is released and the doc is updated).
+
+  # How to use
+
+    Launch the action called 'OneSmallStep' (other actions are provided but we'll get on this later). You should now see OSS's main dialog - One Small Step is active (it is active as long as this dialog is visible). At the top of it, the name of the target MIDI track / item / take will be displayed if there's one eligible that matches your current selection. It is important to note that the track should be armed for record (OSS will give you an indication if you forgot to arm the recording). If everyhing's ready, a red circle will glow, meaning that in this configuration, OneSmallStep is able to do its job (listen to MIDI events, and step input/patch the current MIDI item).
+
+  ## Input modes
+
+    You can then select your input mode between Off / Keyboard / Sustain Pedal / Action.
+
+  ### Off mode
+
+    Does nothing. It just allows to keep the window open and play with the params without the risk to edit something.
+
+  ### Keyboard
+
+    Notes are added to the MIDI item at the current position, when the keys are released.
+
+  ### Sustain Pedal
+
+    Hold keys on your MIDI controller, then press the sustain pedal to validate them. This is convenient when playing with chords for example.
+
+  ### Action
+
+    It's the same thing as with the sustain pedal, except that held notes are validated with a REAPER action. This action is the one called 'OneSmallStep Commit'. You can assign a shortcut to it, and your shortcut will take the role of the sustain pedal for validation.
+
+  ## Note length
+
+    You can adjust the length of the input notes here.
+
+  ## Other actions
+
+    To speed up your flow, multiple actions are provided to quickly change OSS parameters, so that you can assign shortcuts to them. Those are the "Change note len", "Decrease/Increase note len", "Toggle note len modifier" actions, whose names should be safe explanatory. The "Cleanup helper JSFXs" is here for cleaniness, to remove the Helper JSFXs that are installed automatically on the input FX chain of your tracks when OSS is running (it could have been done automatically when closing the tool, but it adds an entry in the undo stack, which is annoying, and I don't have a solution for this yet).
+
+  # Toolbar icons
+
+    Two toolbar icons are provided with OSS, one icon for launching OSS ('toolbar_one_small_step'), and one for launching the cleanup script ('toolbar_one_small_step_cleanup').
+
+  # Credits
+
+    This tool takes a lot of inspiration in tenfour's "tenfour-step" scripts. Epic hail to tenfour for opening the way !
+
+--]]
+
+-------------------------
+
+--[[
+# Ruby script to convert from png > lua to load binary img for ReaImGui
```

Comment:
There's also "Dump file to string literal" in ReaPack for that.

Reply:
Cool ! I didn't have the reflex to check in ReaPack for those little convenient tools. Thanks for the tip !